src/tree/ShallowResourceTreeItem.ts

```diff
@@ -94,12 +94,12 @@ export class ShallowResourceTreeItem extends AzExtTreeItem implements GroupableR
         switch (groupBySetting) {
```

Comment:
What does this class do again?

Reply:
It's for resources that have no extension associated with them. https://github.com/microsoft/vscode-azureresourcegroups/blob/bc86665054c257331e7d6efbdcf4ed28f399a455/src/tree/SubscriptionTreeItem.ts#L55-L66

Reply:
We could have a fallback resolver that handles this scenario built into the host extension. I'm imagining that same resolver could either attach a resolved item that will make the item click show the raw JSON (for things that don't have an extension), or the extension marketplace page (for things that do).

Reply:
Ok, I'm going to merge this and work on that later.

Reply:
Good plan, I was gonna suggest the same.

Reply:
I think it's actually best to not have a fallback resolver for the resources that don't have extensions, and instead have a fallback class. Because our ResolvableTreeItemBase is an AzExtParentTreeItem so it expects to have children. And I don't think we plan on resources without extensions having children.

We could split `ResolvableTreeItemBase` into `ResolvableParentTreeItemBase`, but I think that would make more sense for VMs (resource with an extension that doesn't have children).

Reply:
`hasMoreChildrenImpl` is for implementing the load more action. Is there another way?

> hasMoreChildrenImpl
> Implement this as a part of the "Load more..." action. Should not be called directly
> @returns — 'true' if there are more children and a "Load more..." node should be displayed

Reply:
What's the down side of having it be a parent item though? We may also be able to set the collapsible state to `None` which I think makes it appear non-expandable.

Reply:
Gotcha. I'll double check about `collapsibleState` and make sure it's doing what we want.

Reply:
Ok good, it does do what we want!

With `collapsibleState` set to expanded (which we use by default for compose grouping):
![image](https://user-images.githubusercontent.com/36966225/157746469-8a2bfb2b-f4a5-41af-91cf-33ffbc8fd684.png)

With it set to none:
![image](https://user-images.githubusercontent.com/36966225/157746523-7a4bb8af-e5b1-4d9f-be28-3616d93a762d.png)